build.gradle

```diff
@@ -94,6 +165,46 @@ task currentJar(type: Copy){
 
 shadowJar {
     finalizedBy currentJar
+    from('src/main/resources/properties.templates') {
+        // for the default jar, we want the properties file that enables the picard parser
+        include legacySourcePropertyFile
+        into picardTargetPropertiesPath
+        rename { String fileName ->
+            fileName.replace(legacySourcePropertyFile, picardTargetPropertyFile)
+        }
+    }
+}
+
+// Create picardBarclay.jar, which is a identical to picard.jar, but contains a .properties
+// file that tells Picard to use the Barclay command line parser instead of the Picard
+// command line parser.
+//
+task barclayJar(type: com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar)  {
+    configurations = [project.configurations.runtime]
+    from project.sourceSets.main.output
+    manifest {
+        inheritFrom project.tasks.jar.manifest
+    }
+    from('src/main/resources/properties.templates') {
+        // for the default jar, we want the properties file that enables the picard parser
+        include barclaySourcePropertyFile
+        into picardTargetPropertiesPath
+        rename { String fileName ->
+            fileName.replace(barclaySourcePropertyFile, picardTargetPropertyFile)
+        }
+    }
+    exclude('META-INF/INDEX.LIST', 'META-INF/*.SF', 'META-INF/*.DSA', 'META-INF/*.RSA')
+    archiveName 'picardBarclay.jar'
+}
+
+task barclayTest(type: Test) {
+    // Run all the tests using, but with the command line converter and the Barclay command line parser
+    systemProperty 'picard.convertCommandLine', 'true'
+    systemProperty 'picard.useLegacyParser', 'false'
+}
+
+test {
```

Comment:
Could you add a `legacyTest(type: Test)` target and make test depend on both that and `barclayTest` so that it's possible to run just one or the other.